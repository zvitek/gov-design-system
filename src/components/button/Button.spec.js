import { shallowMount } from '@vue/test-utils'
import GovButton from '@components/button/Button'

let wrapper

describe('GovButton', () => {
    beforeEach(() => {
        wrapper = shallowMount(GovButton)
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('GovButton')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('emit a click event', () => {
        const click = jest.fn()
        wrapper = shallowMount(GovButton, {
            listeners: {
                'click': click
            }
        })
        wrapper.find('.gov-button').trigger('click')
        expect(click).toHaveBeenCalledTimes(1)
    })

    it('should be primary outlined', () => {
        wrapper.setProps({
            primaryOutlined: true
        })
        expect(wrapper.classes()).toContain('gov-button--primary-outlined')
    })

    it('should be inversed', () => {
        wrapper.setProps({
            inversed: true
        })
        expect(wrapper.classes()).toContain('gov-button--inversed')
    })

    it('should be small', () => {
        wrapper.setProps({
            small: true
        })
        expect(wrapper.classes()).toContain('gov-button--small')
    })

    it('should be large', () => {
        wrapper.setProps({
            large: true
        })
        expect(wrapper.classes()).toContain('gov-button--large')
    })

    it('should set tag to "button" if disabled', () => {
        wrapper.setProps({
            tag: 'a'
        })
        expect(wrapper.vm.computedTag).toBe('a')

        wrapper = shallowMount(GovButton, {
            attrs: {
                'disabled': true
            }
        })
        expect(wrapper.vm.computedTag).toBe('button')
    })

    it('should not emit click if disabled', () => {
        const click = jest.fn()
        wrapper = shallowMount(GovButton, {
            listeners: {
                'click': click
            },
            attrs: {
                'disabled': true
            }
        })
        wrapper.find('.gov-button').trigger('click')
        expect(click).toHaveBeenCalledTimes(0)
    })
})
