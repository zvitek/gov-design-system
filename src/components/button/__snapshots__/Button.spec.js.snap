// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovButton render correctly 1`] = `
<button type="button" class="gov-button">
    <!----></button>
`;
